calculatorUICOOP/ViewModels/MainPageViewModel.cs

```diff
@@ -106,28 +118,53 @@ public void DeleteLastInput()
             {
                 ClearScreen();
             }
-           
+
         }
 
         public void ShowRemainderOnDisplay(string remainder)
         {
-
+            AssignOperator(remainder);
         }
 
         public void ShowDecimalOnDisplay(string decimalDot)
         {
-
+            if (!_hasDecimal)
+            {
+                DisplayContent += decimalDot;
+                _hasDecimal = true;
+            }
         }
 
         public void Equals()
         {
-
+            double _number2 = Convert.ToDouble(DisplayContent);
+            switch (_operator)
```

Comment:
To reduce the repeated statement of Convert.ToDouble(DisplayContent), you should assign a local variable to it here since it is converted in every case statement.

Reply:
Needs clarification. Resolving.